src/components/base/Client.ts

```diff
@@ -26,11 +29,12 @@ export class BaseClient implements Base {
     request.setDistanceMm(distanceMm)
     request.setExtra(googleProtobufStructPb.Struct.fromJavaScript(extra))
 
-    const response = await promisify<baseApi.MoveStraightRequest, baseApi.MoveStraightResponse>(
+    this.options.requestLogger?.(request)
```

Comment:
Im curious, what happens when requestLogger is not set and this line is called. will it break? or just not run?

Reply:
the `?.()` syntax makes it so we only execute the call if requestLogger is set

Reply:
but testing on controls page now to verify

Reply:
follow up question, has this been tested in local? and do we need any .bind to make this work?

Reply:
looks good in local, no need to bind here